Setup test suite

joneshf-cn opened the pull request:
##### What it does

Pulls the test JSON values out to the test suite.

##### Still to do

* [x] Move the JSON from `Data.Expr`.

joneshf-cn committed:
Move the test JSON to the test suite

The JSON we had here was a great test case.
We make it a legit case by moving it to the test suite.

This also means we need to make up a real `Page _` value.
We do that, and make a changes in the process.
The `key` fields were all the same initially.
It seems like we want them to be unique per `Field _`.

joneshf-cn committed:
Fix `key` for `active`

Turns out the last commit was kind of a lie.
We only sort of fixed the issue with keys.

We might want to think about how to actually make these unique
across the whole form.
Might be a bit too complex and we should just be more careful.

joneshf-cn committed:
Merge remote-tracking branch 'origin/master' into HEAD

joneshf-cn committed:
Move suite out to its own module

We're about to move more stuff into the test suite.
We can start organizing the code a little bit so it's easier later on.

We also namespace things properly.
The filename reflects the module name.
This should also make things a bit easier to find later on.

joneshf-cn committed:
Extract JSON to test suite

Another good set of tests to put in the suite.

While we did write `assertRight` again,
we probably are fine since it's simple enough.
We might think about pushing this upstream to `purescript-test-unit`.
Surely other people would find it (or something similar useful).

davezuch commented on src/Data/Form.purs:
```diff
@@ -185,69 +185,73 @@ instance decodeInputSource :: DecodeJson InputSource where
-  , "default": """ <> val false "Boolean" <> """
+firstName :: Field Expr
+firstName =
+  { name: mkExpr (Val identity "First Name")
```
I know this test data isn't meant to stick around, but while we're manually constructing these, it might be worthwhile adding a helper function so we can just do something like
```purescript
  name: val_ "First Name"
```

joneshf-cn replied:
Good idea! I went ahead and made a helper for all the cases since we're most likely going to be writing them once we build momentum.

joneshf-cn committed:
Create a couple of helper values

Constructing `Expr _`s by hand isn't the most straightforward thing.
We can make it a bit easier to build up `Expr _`s.
We make helpers for all cases of `ExprF _ _`,
even though we currently only use `Val`s.

davezuch reviewed (approved):
Nice, gonna merge this
